main.go

```diff
@@ -108,6 +113,17 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "Gatekeeper")
 		os.Exit(1)
 	}
+	if err = (&controllers.ConstraintStatusReconciler{
```

Comment:
Could you use a `Transform` function so that you are only caching the fields you need from ConstraintPodStatus to save memory?
https://github.com/open-cluster-management-io/governance-policy-framework-addon/blob/f0e061c0ef033fb41aafe801b48753ad219c6def/main.go#L411-L428

You could also try to limit the cache/reconciles to just the audit ConstraintPodStatus objects with a filter similar to this, though I don't know if that field would be filterable like that. Probably not.
https://github.com/open-cluster-management-io/governance-policy-framework-addon/blob/f0e061c0ef033fb41aafe801b48753ad219c6def/main.go#L407